site/pages/concepts/middleware.mdx

```diff
@@ -160,6 +160,61 @@ app.frame('/foo', (c) => {/* ... */})
 app.frame('/foo/bar', (c) => {/* ... */})
 ```
 
+### Example: TransactionResponse
+
+You can also write a custom middleware to add additional params to Frog's TransactionResponse (mimics [`Transaction Specs API`](https://warpcast.notion.site/Frame-Transactions-Public-9d9f9f4f527249519a41bd8d16165f73#8235c261091e4499b417e71dd1753e05)).
+
+```tsx twoslash
+// @noErrors
+import { Frog } from 'frog'
+ 
+export const app = new Frog()
+
+app.transaction( // [!code focus]
+  '/mint', // [!code focus]
+  async (c, next) => { // [!code focus]
+    await next(); // [!code focus]
+    const txParams = await c.res.json(); // [!code focus]
+    txParams.attribution = false; // [!code focus]
+    c.res = new Response(JSON.stringify(txParams), { // [!code focus]
+      headers: { // [!code focus]
+        'Content-Type': 'application/json', // [!code focus]
+      }, // [!code focus]
+    }); // [!code focus]
+  }, // [!code focus]
+  (c) => {/* ... */} // [!code focus]
+); // [!code focus]
```

Comment:
the provided example is indeed generic but probably not very DX friendly.

I'd rather see `attribution` flag added (#172), as this looks as more like a feature frog should support internally.